libraries/chain/database_worker_objects.cpp

```diff
@@ -246,4 +246,38 @@ namespace golos { namespace chain {
         });
     }
 
+    void database::set_clear_old_worker_techspec_approves(bool clear_old_worker_techspec_approves) {
+        _clear_old_worker_techspec_approves = clear_old_worker_techspec_approves;
+    }
+
+    bool database::clear_old_worker_techspec_approves() const {
+        return _clear_old_worker_techspec_approves;
+    }
+
+    void database::clear_expired_worker_objects() {
+        if (has_hardfork(STEEMIT_HARDFORK_0_21__1013)) {
+            return;
+        }
+
+        const auto& mprops = get_witness_schedule_object().median_props;
+
+        const auto& idx = get_index<worker_techspec_index>().indices().get<by_created>();
+        auto itr = idx.begin();
+        while (itr != idx.end() && itr->created + mprops.worker_techspec_approve_term_sec <= head_block_time()) {
+            if (_clear_old_worker_techspec_approves) {
+                const auto& wtao_idx = get_index<worker_techspec_approve_index, by_techspec_approver>();
+                auto wtao_itr = wtao_idx.find(std::make_tuple(itr->author, itr->permlink));
+                while (wtao_itr != wtao_idx.end() && wtao_itr->author == itr->author && wtao_itr->permlink == itr->permlink) {
+                    const auto& wtao = *wtao_itr;
+                    ++wtao_itr;
+                    remove(wtao);
+                }
+            }
```

Comment:
Copy-paste of lambda-function `clear_approves()` in `worker_techspec_approve_evaluator::do_apply()`